lib/suite.js

```diff
@@ -0,0 +1,126 @@
+   *
+   * Expands suites by their (n) via the `chunks` property to multiple suites.
+   */
+const prepareSuitesChunks = (preferredSuites, suites, config) => {
+  Object.entries(suites).forEach((suite) => {
+    const [suiteName, suiteConfig] = suite;
+    const pattern = suite.tests || config.tests;
+
+    if (!suiteConfig.chunks || !Number.isFinite(suiteConfig.chunks) || !pattern) {
+      return;
+    }
+
+    delete suites[suiteName];
+    chunk.createChunks(suiteConfig, pattern).forEach((suiteChunkConfig, index) => {
+      suites[`${suiteName}:chunk${index + 1}`] = suiteChunkConfig;
+    });
+  });
+
+  return suites;
+};
+
+  /**
+   *
+   * @param array preferredSuites Format: ['suite1', 'suite2', ...]
+   * @param object suites Format: {suite1: {}, suite2: {}, ... }
+   * @param object config
+   *
+   * Expands browser declared via `browsers` property to multiple
+   * suites that all have one single `browser` property and omits
+   * the `browsers` property.
+   */
+const prepareSuitesBrowsers = (preferredSuites, suites, config) => {
+  Object.entries(suites).forEach((suite) => {
+    const [suiteName, suiteConfig] = suite;
+    const browsers = [];
+
+    delete suites[suiteName];
+    suiteConfig.browsers.forEach((browser) => {
+      const browserConfig = browser.browser ? browser : { browser };
+      const suiteBrowserConfig = { ...suiteConfig, browser: browserConfig };
+
+      browsers.push(browserConfig.browser);
+      const count = browsers.filter((b) => {
+        return b === browserConfig.browser;
+      }).length;
+
+      delete suiteBrowserConfig.browsers;
+      suites[`${suiteName}:${browserConfig.browser}${count}`] = suiteBrowserConfig;
+    });
+  });
+
+  return suites;
+};
+
+  /**
+   *
+   * @param array preferredSuites Format: ['suite1', 'suite2', ...]
+   * @param object suites Format: {suite1: {}, suite2: {}, ... }
+   * @param object config
+   *
+   * Filters all suites by their `browser` property. The propoerty `browsers` is ignored.
+   * If value of property `browser` does not match the preferred `browser` in conjugation
+   * with the preferredSuiteName then the suite is removed from configuration.
+   */
+const filterPreferredBrowsers = (preferredSuites, suites, config) => {
+  preferredSuites.forEach((preferredSuite) => {
+    const [preferredSuiteName, preferredSuiteBrowserName] = preferredSuite.split(':');
+
+    preferredSuites.forEach((preferredSuite) => {
+      const [preferredSuiteName, preferredSuiteBrowserName] = preferredSuite.split(':');
+      Object.entries(suites).forEach((suite) => {
+        const [suiteName, suiteConfig] = suite;
+        const suiteBrowserName = suiteConfig.browser.browser;
+
+        if (preferredSuiteBrowserName) {
+          if (preferredSuiteName === suiteConfig.parentSuiteName && preferredSuiteBrowserName !== suiteBrowserName) {
+            delete suites[suiteName];
+          }
+        }
+      });
+    });
+  });
+
+  return suites;
+};
```

Comment:
It looks to me this should be refactored as a class. That adds more readability to the code. 
`preferredSuites, suites, config` params are the same for all functions so I'd better add them to constructor of a class and make them functions inside it.

Reply:
Yes, I’d like to introduce a collection object that holds the items including “suite” config, also manages add and remove. 

Second I’d like to have the prepare* and filter* functions as callbacks on that collection.

Reply:
Ok, let's call that `Collection`!